core/src/main/java/com/sequenceiq/cloudbreak/service/sharedservice/DatalakeService.java

```diff
@@ -0,0 +1,265 @@
+            }
+        }
+        stackResponse.setSharedService(sharedServiceResponse);
+    }
+
+    public Optional<Stack> getDatalakeStackByDatahubStack(Stack datahubStack) {
+        Long datalakeResourceId = datahubStack.getDatalakeResourceId();
+        if (datalakeResourceId != null) {
+            Optional<DatalakeResources> datalakeResource = datalakeResourcesService.findById(datalakeResourceId);
+            if (datalakeResource.isPresent() && datalakeResource.get().getDatalakeStackId() != null) {
+                Long datalakeStackId = datalakeResource.get().getDatalakeStackId();
+                Stack datalakeStack = stackService.getByIdWithListsInTransaction(datalakeStackId);
+                return Optional.of(datalakeStack);
+            }
+        }
+        LOGGER.info("There is no datalake resource id has set for the cluster.");
+        return Optional.empty();
+    }
+
+    public Optional<Stack> getDatalakeStackByStackEnvironmentCrn(Stack datahubStack) {
+        Set<DatalakeResources> datalakeResources = datalakeResourcesService
+                .findDatalakeResourcesByWorkspaceAndEnvironment(datahubStack.getWorkspace().getId(), datahubStack.getEnvironmentCrn());
+        Optional<Stack> datalakeStack = Optional.ofNullable(datalakeResources)
+                .map(Set::stream).flatMap(Stream::findFirst)
+                .map(DatalakeResources::getDatalakeStackId)
+                .map(stackService::getByIdWithListsInTransaction);
+
+        return datalakeStack;
+    }
+
+    public Long getDatalakeResourceId(Long datalakeStackId) {
+        Optional<DatalakeResources> datalakeResources = datalakeResourcesService.findByDatalakeStackId(datalakeStackId);
+        if (datalakeResources.isPresent()) {
+            return datalakeResources.get().getId();
+        }
+        return null;
+    }
+
+    public SharedServiceConfigsView createSharedServiceConfigs(Stack source) {
+        Cluster cluster = source.getCluster();
+        Optional<DatalakeResources> datalakeResources = getDataLakeResource(source);
+        return createSharedServiceConfigs(cluster.getBlueprint(), cluster.getPassword(), datalakeResources);
+    }
+
+    public String getDatalakeCrn(StackV4Request source, Workspace workspace) {
+        if (source.getSharedService() != null && isNotBlank(source.getSharedService().getDatalakeName())) {
+            Optional<Stack> result = stackService
+                    .findStackByNameOrCrnAndWorkspaceId(NameOrCrn.ofName(source.getSharedService().getDatalakeName()), workspace.getId());
+            if (result.isPresent()) {
+                return result.get().getResourceCrn();
+            } else {
+                LOGGER.debug("No datalake resource found for data lake: {}", source.getSharedService().getDatalakeName());
+            }
+        }
+        return null;
+    }
+
+    public void setDatalakeIdOnStack(Stack stack, StackV4Request source, Workspace workspace) {
```

Comment:
the setting could be pushed to the side where this method is called (e.g. the other methods returning values, this is setting a stack property)